docs/provider/beyondtrust.md

```diff
@@ -0,0 +1,124 @@
+## BeyondTrust Password Safe
+
+External Secrets Operator integrates with [BeyondTrust Password Safe](https://www.beyondtrust.com/docs/beyondinsight-password-safe/).
+
+Warning: The External Secrets Operator is designed to write secrets to Kubernetes secrets which are by default written to etcd base 64 encoded. This is not secure, you must configure Kubernetes encryption or use third-party encryption for production environments.
+
+Warning: If the BT provider secret is deleted it will still exist in the Kubernetes secrets.
```

Comment:
There is a feature for `DeletionPolicy: Delete` which prevents this from happening. Perhaps it's worth to implement it since you want to warn users explicitly?

Reply:
I was checking that and I see `DeletionPolicy: Delete` option is just used on `kind: PushSecret,` and current provider is not going to used that kind.

Reply:
I think Gustavo is talking about this:
```go

// ExternalSecretDeletionPolicy defines rules on how to delete the resulting Secret.
// +kubebuilder:validation:Enum=Delete;Merge;Retain
type ExternalSecretDeletionPolicy string

const (
	// Delete deletes the secret if all provider secrets are deleted.
	// If a secret gets deleted on the provider side and is not accessible
	// anymore this is not considered an error and the ExternalSecret
	// does not go into SecretSyncedError status.
	DeletionPolicyDelete ExternalSecretDeletionPolicy = "Delete"

	// Merge removes keys in the secret, but not the secret itself.
	// If a secret gets deleted on the provider side and is not accessible
	// anymore this is not considered an error and the ExternalSecret
	// does not go into SecretSyncedError status.
	DeletionPolicyMerge ExternalSecretDeletionPolicy = "Merge"

	// Retain will retain the secret if all provider secrets have been deleted.
	// If a provider secret does not exist the ExternalSecret gets into the
	// SecretSyncedError status.
	DeletionPolicyRetain ExternalSecretDeletionPolicy = "Retain"
)
```

Reply:
Could you please point me out where this is being implemented ? or give an example, I have not got the point about how we can implemented that, sorry.